components/locid/examples/canonicalize_locales.rs

```diff
@@ -0,0 +1,47 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/master/LICENSE ).
+// A sample application which takes a comma separated list of locales,
+// makes them canonical and serializes the list back into a comma separated list.
+
+use std::env;
+
+use icu_locid::Locale;
+
+const DEFAULT_INPUT: &str = "sr-cyrL-rS, es-mx, und-arab-u-ca-Buddhist";
+
+fn canonicalize_locales(input: &str) -> String {
+    // Split input string and canonicalize each locale identifier.
+    let canonical_locales: Vec<String> = input
+        .split(',')
+        .filter_map(|s| Locale::canonicalize(s.trim()).ok())
+        .collect();
+
+    canonical_locales.join(", ")
+}
+
+fn main() {
+    let args: Vec<String> = env::args().collect();
+
+    let input = if let Some(input) = args.get(1) {
+        input.as_str()
+    } else {
+        DEFAULT_INPUT
+    };
+    let _output = canonicalize_locales(&input);
+
+    #[cfg(debug_assertions)]
+    println!("\nInput: {}\nOutput: {}", input, _output);
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    const DEFAULT_OUTPUT: &str = "sr-Cyrl-RS, es-MX, und-Arab-u-ca-buddhist";
```

Comment:
You have DEFAULT_OUTPUT twice

Actually, I'd probably always hardcode the list. Why does it need to be a CLI argument?

Reply:
My argument *for* the list and to make it optional is that it makes the example very... testible? playable? Not sure what's the right word.

It makes it easy for someone to launch it with different arguments and see what happens, which is a good mental model for exploration.

Reply:
There is one DEFAULT_INPUT and one DEFAULT_OUTPUT.

I was following the other example, where you can enter locale to canonicalize in CLI. We can remove CLI when #[start] syntax becomes stable.

Reply:
Actually, having the CLI option let me play with the example - that's how I found the -u- bug in #368